lib/hammer_cli_katello/host_collection.rb

```diff
@@ -51,15 +51,47 @@ class InfoCommand < HammerCLIKatello::InfoCommand
     end
 
     class HostsCommand < HammerCLIKatello::ListCommand
-      resource :host_collections, :hosts
+      resource :hosts, :index
       command_name "hosts"
 
+      option "--id", "HOST_COLLECTION_ID", _("Host Collection ID")
+      option "--name", "HOST_COLLECTION_NAME", _("Host Collection Name")
+
+      validate_options do
+        host_collection_options = [:option_id, :option_name]
+        organization_options = [:option_organization_id, :option_organization_name,
+                                :option_organization_label]
+
+        any(*host_collection_options).required
+
+        if option(:option_name).exist?
+          any(*organization_options).required
+        end
+      end
+
+      def host_collection_options
+        {
+          "option_name" => option_name,
+          "option_organization_name" => option_organization_name,
+          "option_organization_id" => option_organization_id,
+          "option_organization_label" => option_organization_label
+        }
+      end
+
+      def request_params
+        params = {}
```

Comment:
I think you can set the attribute_name when defining the option, like [this](https://github.com/Katello/hammer-cli-katello/blob/364397877e80cf99d098748bb418476f1fea079a/lib/hammer_cli_katello/content_view_version.rb#L76). This allows `--name` to be given to hammer as `option_host_collection_name`